pkg/detector/detector.go

```diff
@@ -1339,14 +1343,47 @@ func (d *ResourceDetector) HandleClusterPropagationPolicyCreationOrUpdate(policy
 		d.Processor.Add(key)
 	}
 
-	// if preemption is enabled, handle the preemption process.
+	// If preemption is enabled, handle the preemption process.
+	// If this policy succeeds in preempting resource managed by other policy, the label marked on relevant resource will be replaced,
+	// which gives the resource template a change to match to this policy.
+	// However, if PolicyLazyReconcile feature enabled, the marked label will also be replaced, but the label change is by Karmada self,
+	// thus the event will be ignored by Detector, so the resource will not adjust to this policy immediately util the resource itself updated.
 	if preemptionEnabled(policy.Spec.Preemption) {
 		return d.handleClusterPropagationPolicyPreemption(policy)
 	}
 
 	return nil
 }
 
+// addRelatedTemplateToProcessorQueue When updating fields other than ResourceSelector, should first find the corresponding
+// ResourceBinding/ClusterResourceBinding and add the bound object to the processor's queue for reconciliation to make sure that
+// ClusterPropagationPolicy's updates can be synchronized to ResourceBinding/ClusterResourceBinding.
+func (d *ResourceDetector) addRelatedTemplateToProcessorQueue(policy *policyv1alpha1.ClusterPropagationPolicy) error {
+	resourceBindings, err := d.listCPPDerivedRB(policy.Name)
+	if err != nil {
+		return fmt.Errorf("failed to list CPP(%s) derived RB: %+v", policy.Name, err)
+	}
+	clusterResourceBindings, err := d.listCPPDerivedCRB(policy.Name)
+	if err != nil {
+		return fmt.Errorf("failed to list CPP(%s) derived CRB: %+v", policy.Name, err)
+	}
+	for _, rb := range resourceBindings.Items {
+		resourceKey, err := helper.ConstructClusterWideKey(rb.Spec.Resource)
+		if err != nil {
+			return fmt.Errorf("failed to get resource ClusterWideKey from RB(%s): %+v", rb.Name, err)
+		}
+		d.Processor.Add(resourceKey)
+	}
+	for _, crb := range clusterResourceBindings.Items {
+		resourceKey, err := helper.ConstructClusterWideKey(crb.Spec.Resource)
+		if err != nil {
+			return fmt.Errorf("failed to get resource ClusterWideKey from CRB(%s): %+v", crb.Name, err)
+		}
+		d.Processor.Add(resourceKey)
+	}
+	return nil
+}
+
```

Comment:
The reason for extracting this function is that the complexity of the code in the original function exceeded limits, causing ci-lint to crash.